### ISSUE TITLE
[Troubleshoot]: Unable to reproduce maniskill_ppo_openvla result

### ISSUE THREAD

**Issue by zixiao-bios** [RLinf/RLinf#133]
### Problem description

Can not reproducing the maniskill_ppo_openvla training reault with raw config.
The success rate was rapidly reduced from 0.5 to 0:
<img width="730" height="297" alt="Image" src="https://github.com/user-attachments/assets/2a1a2c67-1d99-43cd-8695-24044cf87215" />

<img width="1465" height="2216" alt="Image" src="https://github.com/user-attachments/assets/648dbe75-34a9-4899-ad82-ba3e74093545" />

### Environment

Python 3.11.10
arm_pytorch_utilities 0.4.3
array_record 0.8.1
pytorch-kinematics 0.7.5
pytorch-seed 0.2.0
ray 2.47.0
torch 2.5.1+cu124
torch_memory_saver 0.0.8
torchaudio 2.5.1+cu124
torchdata 0.11.0
torchelastic 0.2.2
torchvision 0.20.1+cu124
transformers 4.40.1
Wed Sep 24 16:49:58 2025
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.124.06 Driver Version: 570.124.06 CUDA Version: 12.8 |
|-----------------------------------------+------------------------+----------------------+
| GPU Name Persistence-M | Bus-Id Disp.A | Volatile Uncorr. ECC |
| Fan Temp Perf Pwr:Usage/Cap | Memory-Usage | GPU-Util Compute M. |
| | | MIG M. |
|=========================================+========================+======================|
| 0 NVIDIA H100 80GB HBM3 On | 00000000:18:00.0 Off | 0 |
| N/A 27C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 1 NVIDIA H100 80GB HBM3 On | 00000000:2A:00.0 Off | 0 |
| N/A 28C P0 74W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 2 NVIDIA H100 80GB HBM3 On | 00000000:3A:00.0 Off | 0 |
| N/A 29C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 3 NVIDIA H100 80GB HBM3 On | 00000000:5D:00.0 Off | 0 |
| N/A 28C P0 70W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 4 NVIDIA H100 80GB HBM3 On | 00000000:84:00.0 Off | 0 |
| N/A 27C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 5 NVIDIA H100 80GB HBM3 On | 00000000:8B:00.0 Off | 0 |
| N/A 29C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 6 NVIDIA H100 80GB HBM3 On | 00000000:91:00.0 Off | 0 |
| N/A 28C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+
| 7 NVIDIA H100 80GB HBM3 On | 00000000:E4:00.0 Off | 0 |
| N/A 27C P0 69W / 700W | 1MiB / 81559MiB | 0% Default |
| | | Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes: |
| GPU GI CI PID Type Process name GPU Memory |
| ID ID Usage |
|=========================================================================================|
| No running processes found |
+-----------------------------------------------------------------------------------------+

### Reproduction

1. Use docker image: rlinf/rlinf:agentic-openvla-rlinf0.1-torch2.5.1
2. Use raw config `examples/embodiment/config/maniskill_ppo_openvla.yaml`
3. Just modify:
    - env/train, env/eval -> PutOnPlateInScene25Main
    - model_dir, checkpoint_load_path, tokenizer_model -> `/path/to/my/rl4vla/openvla-7b-rlvla-warmup/`

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by zixiao-bios** [comment]
My entire config:
```
{
    "env": {
        "train": {
            "simulator_type": "maniskill",
            "auto_reset": true,
            "ignore_terminations": true,
            "use_rel_reward": true,
            "seed": 0,
            "num_group": 15,
            "group_size": 1,
            "use_fixed_reset_state_ids": false,
            "only_eval": false,
            "max_episode_steps": 80,
            "video_cfg": {
                "save_video": true,
                "info_on_video": true,
                "video_base_dir": "/RLinf-raw/logs/20250925-06:05:21/video/train"
            },
            "init_params": {
                "id": "PutOnPlateInScene25Main-v3",
                "num_envs": 15,
                "obs_mode": "rgb+segmentation",
                "control_mode": "arm_pd_ee_target_delta_pose_align2_gripper_pd_joint_pos",
                "sim_backend": "gpu",
                "sim_config": {
                    "sim_freq": 500,
                    "control_freq": 5
                },
                "max_episode_steps": 80,
                "sensor_configs": {
                    "shader_pack": "default"
                },
                "render_mode": "all",
                "obj_set": "train"
            }
        },
        "eval": {
            "simulator_type": "maniskill",
            "auto_reset": true,
            "ignore_terminations": true,
            "use_rel_reward": true,
            "seed": 0,
            "num_group": 1,
            "group_size": 1,
            "use_fixed_reset_state_ids": true,
            "num_envs": 1,
            "only_eval": true,
            "video_cfg": {
                "save_video": true,
                "info_on_video": true,
                "video_base_dir": "/RLinf-raw/logs/20250925-06:05:21/video/eval"
            },
            "init_params": {
                "id": "PutOnPlateInScene25Main-v3",
                "num_envs": 4,
                "obs_mode": 1,
                "control_mode": "arm_pd_ee_target_delta_pose_align2_gripper_pd_joint_pos",
                "sim_backend": "gpu",
                "sim_config": {
                    "sim_freq": 500,
                    "control_freq": 5
                },
                "max_episode_steps": 80,
                "sensor_configs": {
                    "shader_pack": "default"
                },
                "render_mode": "all",
                "obj_set": "test"
            }
        },
        "group_name": "EnvGroup",
        "channel": {
            "name": "env_buffer_list",
            "queue_name": "obs_buffer",
            "queue_size": 0
        },
        "enable_offload": false
    },
    "cluster": {
        "num_nodes": 1,
        "component_placement": {
            "actor": "0-7",
            "env": "0-3",
            "rollout": "4-7"
        }
    },
    "runner": {
        "task_type": "embodied",
        "logger": {
            "log_path": "/RLinf-raw/logs/20250925-06:05:21",
            "project_name": "rlinf",
            "experiment_name": "test_openvla",
            "logger_backends": [
                "tensorboard"
            ]
        },
        "max_epochs": 1000,
        "max_steps": -1,
        "only_eval": false,
        "val_check_interval": -1,
        "save_interval": 40,
        "seq_length": 4096,
        "max_prompt_length": 30
    },
    "algorithm": {
        "auto_reset": true,
        "ignore_terminations": true,
        "use_fixed_reset_state_ids": false,
        "require_values": true,
        "shuffle_samples": true,
        "normalize_advantages": true,
        "kl_penalty": "kl",
        "group_size": 1,
        "n_chunk_steps": 80,
        "n_eval_chunk_steps": 80,
        "num_group_envs": 15,
        "rollout_epoch": 1,
        "reward_type": "step_level",
        "logprob_type": "action_level",
        "entropy_type": "action_level",
        "adv_type": "embodied_gae",
        "loss_type": "embodied_ppo",
        "loss_agg_func": "token-mean",
        "kl_beta": 0.0,
        "entropy_bonus": 0,
        "clip_ratio_high": 0.2,
        "clip_ratio_low": 0.2,
        "clip_ratio_c": 3.0,
        "value_clip": 0.2,
        "huber_delta": 10.0,
        "gamma": 0.99,
        "gae_lambda": 0.95,
        "sampling_params": {
            "use_greedy": false,
            "temperature_train": 1.0,
            "temperature_eval": 0.6,
            "top_k": 50,
            "top_p": 1.0,
            "repetition_penalty": 1.0
        },
        "length_params": {
            "max_new_token": 7,
            "max_length": 1024,
            "min_length": 1
        }
    },
    "rollout": {
        "group_name": "RolloutGroup",
        "channel": {
            "name": "env_buffer_list",
            "queue_name": "action_buffer",
            "queue_size": 0
        },
        "mode": "colocate",
        "backend": "huggingface",
        "model_dir": "/inspire/ssd/project/robotsimulation/public/eric/Models/openvla-7b-rlvla-warmup/",
        "enable_offload": true,
        "pipeline_stage_num": 2
    },
    "actor": {
        "group_name": "ActorGroup",
        "channel": {
            "name": "env_buffer_list",
            "queue_name": "replay_buffer",
            "queue_size": 0
        },
        "training_backend": "fsdp",
        "checkpoint_load_path": "/inspire/ssd/project/robotsimulation/public/eric/Models/openvla-7b-rlvla-warmup/",
        "checkpoint_save_path": "../results",
        "micro_batch_size": 20,
        "global_batch_size": 160,
        "seed": 1234,
        "enable_offload": true,
        "model": {
            "model_name": "openvla",
            "action_dim": 7,
            "num_action_chunks": 1,
            "use_proprio": false,
            "unnorm_key": "bridge_orig",
            "center_crop": true,
            "do_sample": false,
            "precision": "bf16",
            "add_bias_linear": false,
            "add_qkv_bias": true,
            "vocab_size": 32000,
            "hidden_size": 4096,
            "policy_setup": "widowx_bridge",
            "vh_mode": "a0",
            "image_size": [
                224,
                224
            ],
            "is_lora": true,
            "lora_rank": 32,
            "ckpt_path": null,
            "attn_implementation": "flash_attention_2",
            "low_cpu_mem_usage": true,
            "trust_remote_code": true,
            "sharding_strategy": "full_shard"
        },
        "optim": {
            "lr": 0.0001,
            "value_lr": 0.003,
            "adam_beta1": 0.9,
            "adam_beta2": 0.95,
            "adam_eps": 1e-05,
            "clip_grad": 1.0
        },
        "tokenizer": {
            "tokenizer_type": "HuggingFaceTokenizer",
            "tokenizer_model": "/inspire/ssd/project/robotsimulation/public/eric/Models/openvla-7b-rlvla-warmup/",
            "use_fast": false,
            "trust_remote_code": true,
            "padding_side": "right"
        }
    },
    "reward": {
        "use_reward_model": false
    },
    "critic": {
        "use_critic_model": false
    }
}
```

**Comment by zixiao-bios** [comment]
This is the first output video, I noticed that even though the task was successfully executed, the environment was not auto reset. Is this the expected behavior?

https://github.com/user-attachments/assets/f7961d71-8ee2-49d5-a092-fcf8dd137b2e

**Comment by zanghz21** [comment]
Could you please confirm whether the config you used corresponds to the curve you reported? In our experiments, using the same config, we did not observe the same phenomenon—the success rate increased as expected.
Note that we have not yet uploaded the maniskill-ppo YAMLs for the PutOnPlateInScene25Main task. For more stable results, you can try setting ``algorithm.num_group_envs`` to 256 and ``global_batch_size`` to 640. A complete example YAML is attached at the end.

There is, however, a known issue with environment offloading. We have observed cases where the success rate drops dramatically at the second step. We are working on a fix, but for now, please set ``env.enable_offload=False``. Sorry for the inconvenience.

Regarding the behavior in the video: this is expected. In your YAML, you set ``ignore_terminations=True``, which means the environment only ends when it reaches ``max_episode_steps``. If you set ``ignore_terminations=False``, then the env will reset immediately if it reaches the end. If so, ``n_chunk_steps`` may need to be doubled so that the rollout buffer mainly contains completed trajectories.


``` 
defaults:
  - env/train: PutOnPlateInScene25Main
  - env/eval: maniskill_ood_template
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  num_gpus_per_node: 8
  component_placement:
    actor: 0-7
    env: 0-7
    rollout: 0-7

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "test_openvla"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1

  only_eval: False
  val_check_interval: -1
  save_interval: 40
  seq_length: 4096
  max_prompt_length: 30

algorithm:
  auto_reset: True
  ignore_terminations: True
  use_fixed_reset_state_ids: False
  require_values: True
  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 1
  n_chunk_steps: 80
  n_eval_chunk_steps: 80
  num_group_envs: 256
  rollout_epoch: 1
  reward_type: step_level
  logprob_type: action_level
  entropy_type: action_level

  adv_type: "embodied_gae"
  loss_type: "embodied_ppo"
  loss_agg_func: "token-mean"
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  # params for rollout
  sampling_params:
    use_greedy: False
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 0
    top_p: 1.0
    repetition_penalty: 1.0

  # length argument for autoregressive sampling
  # max length means max amount of tokens to generate
  length_params:
    max_new_token: 7
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"
  channel:
    name: "env_buffer_list"
    queue_name: "obs_buffer"
    queue_size: 0
  enable_offload: False

rollout:
  group_name: "RolloutGroup"
  channel:
    name: ${env.channel.name}
    queue_name: "action_buffer"
    queue_size: 0
  mode: "colocate"
  backend: "huggingface"
  model_dir: "/path/to/rl4vla/openvla-7b-rlvla-warmup/"
  enable_offload: True
  pipeline_stage_num: 1

actor:
  group_name: "ActorGroup"
  channel:
    name: ${env.channel.name}
    queue_name: "replay_buffer"
    queue_size: 0
  training_backend: "fsdp"
  checkpoint_load_path: "/path/to/rl4vla/openvla-7b-rlvla-warmup/"
  checkpoint_save_path: "../results"
  micro_batch_size: 40
  global_batch_size: 640
  seed: 1234
  enable_offload: True

  model:
    model_name: "openvla"
    action_dim: 7
    num_action_chunks: 1
    use_proprio: False
    unnorm_key: bridge_orig
    center_crop: True
    do_sample: False
    
    precision: "bf16"
    add_bias_linear: False
    add_qkv_bias: True
    vocab_size: 32000
    hidden_size: 4096
    policy_setup: "widowx_bridge"
    vh_mode: "a0"
    is_lora: True
    lora_rank: 32
    ckpt_path: null
    attn_implementation: "flash_attention_2"
    low_cpu_mem_usage: True
    trust_remote_code: True

  optim:
    lr: 1.0e-4
    value_lr: 3.0e-3
    adam_beta1: 0.9
    adam_beta2: 0.999
    adam_eps: 1.0e-05
    clip_grad: 10.0

  tokenizer:
    tokenizer_type: "HuggingFaceTokenizer"
    tokenizer_model: "/path/to/rl4vla/openvla-7b-rlvla-warmup/"
    use_fast: False
    trust_remote_code: True
    padding_side: "right"

reward:
  use_reward_model: False

critic:
  use_critic_model: False
```

**Comment by zoeyuchao** [comment]
@zixiao-bios Hi, have u solved this issue？
